### PR TITLE
Implement login via Slack.

### LINKED ISSUE #6: Login via Slack
For various reasons, it's nice to be able to llnk the user to their R4DS Slack account. Add the login with Slack logic from mentordash, and use it to identify the user in the spreadsheet. Use their Slack ID, not just their username, since the username can change.

### PR DESCRIPTION
**Pull request by jonthegeek**
Closes #6.

### CODE REVIEW

**Comment by jonthegeek** [comment]
I'm going to move part of this over to shinyslack and simplify what's here. I wanted to get the PR submitted in case you have time to review, but one more change will either follow this PR or be part of it. Ping me if you review.

File: R/app_server.R [hunk 1]
```diff
@@ -16,6 +16,33 @@
...
+      slack_info$user
+    )
+
+    list(
+      user_name = slack_info$user,
+      user_id = slack_info$user_id
+    )
+  })
+
+  output$username <- shiny::renderText(
```
Comment (priyankagagneja) [inline R/app_server.R:38]: Output display makes sense

File: R/app_server.R [hunk 2]
```diff
@@ -16,6 +16,33 @@
...
 
+  # Display their name so they can log out if it's the wrong person. This should
+  # eventually move into shinyslack as an exported module.
+  slack_user_info <- shiny::reactive({
+    slack_info <- slackcalls::post_slack(
+      slack_method = "auth.test"
+    )
+
+    # Log the current user on shinyapps.
+    message(
```
Comment (priyankagagneja) [inline R/app_server.R:27]: Does this give a pop up saying you are logged in as ? 
Not sure if this needed

File: R/app_server.R [hunk 3]
```diff
@@ -52,7 +79,8 @@
   # Save the user details
   user_info <- shiny::reactive({
     data.frame(
-      user_name = input$username,
+      user_name = slack_user_info()$user_name,
```
Comment (priyankagagneja) [inline R/app_server.R:82]: We probably just need user-id to be saved in the googlesheet/database. The username can just be used to display.

**Review by priyankagagneja — APPROVED** [review]

File: R/app_server.R (hunk 2 above)
Comment (jonthegeek) [reply R/app_server.R:27]: No, it just sends that message to the server, ie it logs it on shinyapps (or in your console if you run_app() locally). Useful for debugging. You can `usethis::pr_fetch(17)` to pull this down locally, and then build it and `run_app()` to see how it works. I also went ahead and deployed at shinyapps ([r4ds.io/bookclubber?book_name=Mastering%20Shiny](r4ds.io/bookclubber?book_name=Mastering%20Shiny) for example).

File: R/app_server.R (hunk 3 above)
Comment (jonthegeek) [reply R/app_server.R:82]: It made it confusing to review the results as a human. We can technically probably look up their username, but it's handy to be able to see it. Once we're in a database, we'll likely have a separate table with the user_id to username translation, but for now it's easiest to have it in one place.

**Pull request merged**